subprojects/docs/src/docs/userguide/authoring-builds/tutorial/partr2_build_lifecycle.adoc

```diff
@@ -0,0 +1,127 @@
+
+Phase 2 - Configuration  :: During this phase, the `Project` objects are configured using the build scripts of all projects which are part of the build are executed.
+
+Phase 3 - Execution  :: Gradle determines the set of tasks, created and configured during the configuration phase, to be executed. Gradle then executes each of the selected tasks.
+
+When Gradle is invoked to execute a task, the lifecycle begins.
+Let's see it in action.
+
+image::gradle-build-lifecycle.png[]
+
+== Step 2. Update the Settings File
+
+Add the following line to the top of the Settings file:
+
+.settings.gradle.kts
+[source, kotlin]
+----
+println("SETTINGS FILE: This is executed during the initialization phase")
+----
+
+== Step 3. Update the Build Script
+
+Add the following lines to the bottom of the Build script:
+
+.app/build.gradle.kts
+[source, kotlin]
+----
+println("BUILD SCRIPT: This is executed during the configuration phase")
+
+tasks.register("task1"){
+    println("REGISTER TASK1: This is executed during the configuration phase")
+}
+
+tasks.register("task2"){
+    println("REGISTER TASK2: This is executed during the configuration phase")
+}
+
+tasks.named("task1"){
+    println("NAMED TASK1: This is executed during the configuration phase")
+    doFirst {
+        println("NAMED TASK1 - doFirst: This is executed during the execution phase")
+    }
+    doLast {
+        println("NAMED TASK1 - doLast: This is executed during the execution phase")
+    }
+}
+
+tasks.named("task2"){
+    println("NAMED TASK2: This is executed during the configuration phase")
+    doFirst {
+        println("NAMED TASK2 - doFirst: This is executed during the execution phase")
+    }
+    doLast {
+        println("NAMED TASK2 - doLast: This is executed during the execution phase")
+    }
+}
+----
+
+== Step 4. Run a Gradle Task
+
+Run the `task1` task that you registered and configured in Step 3:
+
+[source]
+----
+$ ./gradlew task1
+
+SETTINGS FILE: This is executed during the initialization phase     //<1>
+
+> Configure project :app
+BUILD SCRIPT: This is executed during the configuration phase       //<2>
+REGISTER TASK1: This is executed during the configuration phase     //<2>
+NAMED TASK1: This is executed during the configuration phase        //<2>
+
+> Task :app:task1
+NAMED TASK1 - doFirst: This is executed during the execution phase  //<3>
+NAMED TASK1 - doLast: This is executed during the execution phase   //<3>
+
+BUILD SUCCESSFUL in 25s
+5 actionable tasks: 3 executed, 2 up-to-date
+----
+<1> *Initialization*: Gradle executes `settings.gradle.kts` to determine the projects to be built and creates a `Project` object for each one.
+<2> *Configuration*: Gradle configures each project by executing the `build.gradle.kts` files. It resolves dependencies and creates a dependency graph of all the available tasks.
+<3> *Execution*: Gradle executes the tasks passed on the command line as well as any dependent tasks.
```

Comment:
🤔 Saying "Dependent tasks" makes me think of tasks that depend upon the requested task, not tasks that are prerequisites of them.  Maybe saying prerequisite tasks would be clearer.